File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/PostDao.java
package seigneur.gauvain.mycourt.data.local.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;
import io.reactivex.Maybe;
import seigneur.gauvain.mycourt.data.model.Draft;

@Dao
public interface PostDao {

    @Query("SELECT * FROM Draft")
    Maybe<List<Draft>> getAllPost();


    @Query("SELECT * FROM Draft WHERE id IN (:shotId)")
    Maybe<Draft> getShotDraftByShotId(String shotId);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    int updateDraft(Draft draft);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertPost(Draft draft);

    @Query("DELETE FROM draft WHERE id = :id")
    int deletDraftByID(int id);

}

File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/PostDao.kt
package seigneur.gauvain.mycourt.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import io.reactivex.Maybe
import seigneur.gauvain.mycourt.data.model.Draft

@Dao
interface PostDao {

    @get:Query("SELECT * FROM Draft")
    val allPost: Maybe<List<Draft>>

    @Query("SELECT * FROM Draft WHERE id IN (:shotId)")
    fun getShotDraftByShotId(shotId: String): Maybe<Draft>

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateDraft(draft: Draft): Int

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPost(draft: Draft): Long

    @Query("DELETE FROM Draft WHERE draftID = :draftID")
    fun deletDraftByID(draftID: Int): Int

}
